catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -25,34 +73,514 @@ protected String generateCreateTableSql(
+      while (resultSet.next()) {
+        createTableSqlSb.append(resultSet.getString("Create Table"));
+      }
+    }
+
+    String createTableSql = createTableSqlSb.toString();
+
+    if (StringUtils.isEmpty(createTableSql)) {
+      throw new NoSuchTableException(
+          "Table %s does not exist in %s.", tableName, connection.getCatalog());
+    }
+
+    return Collections.unmodifiableMap(DorisUtils.extractPropertiesFromSql(createTableSql));
+  }
+
+  @Override
+  protected List<Index> getIndexes(Connection connection, String databaseName, String tableName)
+      throws SQLException {
+    String sql = String.format("SHOW INDEX FROM `%s` FROM `%s`", tableName, databaseName);
+
+    // get Indexes from SQL
+    try (PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        ResultSet resultSet = preparedStatement.executeQuery()) {
+
+      List<Index> indexes = new ArrayList<>();
+      while (resultSet.next()) {
+        String indexName = resultSet.getString("Key_name");
+        String columnName = resultSet.getString("Column_name");
+        indexes.add(
+            Indexes.of(Index.IndexType.PRIMARY_KEY, indexName, new String[][] {{columnName}}));
+      }
+      return indexes;
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  protected void correctJdbcTableFields(
+      Connection connection, String databaseName, String tableName, JdbcTable.Builder tableBuilder)
+      throws SQLException {
+
+    if (StringUtils.isNotEmpty(tableBuilder.comment())) {
+      return;
+    }
+
+    // Doris Cannot get comment from JDBC 8.x, so we need to get comment from sql
+    String comment = "";
+    String sql =
+        "SELECT TABLE_COMMENT FROM information_schema.TABLES WHERE TABLE_SCHEMA = ? AND TABLE_NAME = ?";
+    try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+      preparedStatement.setString(1, databaseName);
+      preparedStatement.setString(2, tableName);
+
+      try (ResultSet resultSet = preparedStatement.executeQuery()) {
+        while (resultSet.next()) {
+          comment += resultSet.getString("TABLE_COMMENT");
+        }
+      }
+      tableBuilder.withComment(comment);
+    } catch (SQLException e) {
+      throw exceptionMapper.toGravitinoException(e);
+    }
   }
 
   @Override
   protected String generateRenameTableSql(String oldTableName, String newTableName) {
-    throw new UnsupportedOperationException();
+    return String.format("ALTER TABLE `%s` RENAME `%s`", oldTableName, newTableName);
   }
 
   @Override
   protected String generateDropTableSql(String tableName) {
-    throw new UnsupportedOperationException();
+    return String.format("DROP TABLE `%s`", tableName);
   }
 
   @Override
   protected String generatePurgeTableSql(String tableName) {
-    throw new UnsupportedOperationException();
+    return String.format("TRUNCATE TABLE `%s`", tableName);
```

Comment:
`TRUNCATE TABLE` in Doris means dropping all data in the table and retaining the table itself, do it align with our definition of `purgeTable`   @mchades Please help to confirm it.

Reply:
Here should be: truncate + drop = purge

Reply:
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/PURGE.html

https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/DROP-TABLE.html

@mchades I have a little question, what is the exact semantics : `PURGE TABLE xx` or `DROP TABLE xxx PURGE`

Seems there is a little diffrent

Reply:
@zhoukangcn 
- dropTable removes both the metadata and the directory associated with the table from the file system if the table is not an external table. In case of an external table, only the associated metadata is removed.
- purgeTable completely removes both the metadata and the directory associated with the table and skipping trash, if the table is an external table or the catalogs don't support purge table, UnsupportedOperationException is thrown.

For more, please see: https://github.com/datastrato/gravitino/blob/9128db0d2f4bc57dbfb410b51db17887e335cba4/docs/manage-relational-metadata-using-gravitino.md?plain=1#L876C1-L881C127

Reply:
Got it. Thank you very much !

Reply:
I had update the code, throw a UnsupportedOperationException in purge Doris table